src/pdm/cli/commands/publish/__init__.py

```diff
@@ -106,7 +112,11 @@ def _check_response(response: Response) -> None:
             try:
                 response.raise_for_status()
             except requests.HTTPError as err:
-                message = str(err)
+                message = str(err) + "\n"
+                if len(response.text) <= LIMIT_SIZE_RESPONSEDATA:
+                    message += response.text
+                else:
+                    message += response.text[:LIMIT_SIZE_RESPONSEDATA] + "\n...\n(truncated)\n"
```

Comment:
Preferably, on the part of response that is below the limit is fetched at all and this API response limit is enforced in a global/architectural tactic rather than just here.

Reply:
Ah, but you do mean the response text. Will change to logging then.